docs/okeps/okep-5224-connecting-udns/okep-5224-connecting-udns.md

```diff
@@ -0,0 +1,1575 @@
+---
+sh-5.2# ovn-nbctl lr-route-list connect-router-colored-enterprise 
+IPv4 Routes
+Route Table <main>:
+           103.103.0.0/24              192.128.0.12 dst-ip
+           103.103.1.0/24               192.128.0.0 dst-ip
+           103.103.2.0/24               192.128.0.6 dst-ip
+           104.104.0.0/24               192.128.0.4 dst-ip
+           104.104.1.0/24              192.128.0.10 dst-ip
+           104.104.2.0/24              192.128.0.16 dst-ip
+           105.105.0.0/24              192.128.0.14 dst-ip
+           105.105.1.0/24               192.128.0.2 dst-ip
+           105.105.2.0/24               192.128.0.8 dst-ip
+```
+
+**Layer2**
+
+These routes will look like this in the L3 blue, green, yellow sample topology:
+
+* Add logical router policies to `blue_blue.network_transit_router`
+  that steers traffic towards the green and yellow UDN's subnet to connect
+  router on all nodes
+* Add logical router policies to `green_green.network_transit_router`
+  that steers traffic towards the blue and yellow UDN's subnet to connect
+  router on all nodes
+* Add logical router policies to `yellow_yellow.network_transit_router`
+  that steers traffic towards the blue and green UDN's subnet to connect
+  router on all nodes
+* On the `connect-router-colored-enterprise` add routes to specific
+  node subnet slices of the blue, green and yellow transit_routers. We
+  don't need per-node routes like with layer3, since transit router is
+  distributed across all nodes and layer2 is a flat network
+
+```yaml
+sh-5.2# ovn-nbctl lr-policy-list green_green.network_transit_router
+Routing Policies
+      9001 inport == "trtos-green_green.network_ovn_layer2_switch" && ip4.dst == 103.103.0.0/16         reroute               192.128.0.7
+      9001 inport == "trtos-green_green.network_ovn_layer2_switch" && ip4.dst == 105.105.0.0/16         reroute               192.128.0.7
+---
+sh-5.2# ovn-nbctl lr-policy-list blue_blue.network_transit_router
+Routing Policies
+      9001 inport == "trtos-blue_blue.network_ovn_layer2_switch" && ip4.dst == 104.104.0.0/16         reroute              192.128.0.11
+      9001 inport == "trtos-blue_blue.network_ovn_layer2_switch" && ip4.dst == 105.105.0.0/16         reroute              192.128.0.11
+---
+sh-5.2# ovn-nbctl lr-policy-list yellow_yellow.network_transit_router
+Routing Policies
+      9001 inport == "trtos-yellow_yellow.network_ovn_layer2_switch" && ip4.dst == 103.103.0.0/16         reroute               192.128.0.9
+      9001 inport == "trtos-yellow_yellow.network_ovn_layer2_switch" && ip4.dst == 104.104.0.0/16         reroute               192.128.0.9
+---
+sh-5.2# ovn-nbctl lr-route-list connect-router-colored-enterprise
+IPv4 Routes
+Route Table <main>:
+           103.103.0.0/16               192.128.0.4 dst-ip ecmp
+           104.104.0.0/16               192.128.0.0 dst-ip ecmp
+           105.105.0.0/16               192.128.0.2 dst-ip ecmp
```

Comment:
I'm still unsure why these are ECMP?   They'll always be at most a single path.

Reply:
Maybe it's just a remainder of the old version of the okep.  Looking at the mixed topology I see the routes are not ECMP anymore.  In which case we should just remove the "ecmp" keyword here too.